### PR TITLE
Add dynamic reconfigure

### PR DESCRIPTION
**Pull request by efernandez**
Add dynamic reconfigure (only) for calibration parameters.
This is useful to tune this calibration parameters on the fly.

RT-safe implementation achieved on the second commit.

### CODE REVIEW

**Comment by efernandez** [comment]
@paulbovbel

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -298,6 +314,27 @@ namespace diff_drive_controller
 
   void DiffDriveController::update(const ros::Time& time, const ros::Duration& period)
   {
+    // UPDATE DYNAMIC PARAMS
+    // Retreive dynamic params:
+    DynamicParams dynamic_params = *(dynamic_params_.readFromRT());
```
Comment (paulbovbel) [inline diff_drive_controller/src/diff_drive_controller.cpp:319]: This is going to update all these parameters on control cycle, which seems like a lot. Would it be safe to use a boolean flag across threads to notify of update?

Comment (efernandez) [inline diff_drive_controller/src/diff_drive_controller.cpp:319]: The problem is that there's not write from RT methods supported: see [this proposal](https://github.com/ros-controls/realtime_tools/issues/14).

I could add a comment with that link, and we can go back to this discussion when this goes upstream; this is fast enough anyway. What do you think?

**Comment by paulbovbel** [comment]
LGTM, as per prior discussion

File: diff_drive_controller/src/diff_drive_controller.cpp (hunk 1 above)
Comment (efernandez) [inline diff_drive_controller/src/diff_drive_controller.cpp:319]: Comment added. Merging...

**Pull request merged**